services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala

```diff
@@ -31,21 +25,21 @@ object ProjectConfigurationService {
     new ProjectConfigurationService {
 
       def addConfiguration(projectConfiguration: ProjectConfiguration, userId: UserId): Future[Unit] = {
-        def toTuple(config: ProjectFileConfiguration): (Path, Map[String, TypedValue]) =
-          config.path -> config.inputs.map(fileParameter => fileParameter.name -> fileParameter.typedValue).toMap
+        def toTuple(wdlParams: WdlParams): (Path, Map[String, TypedValue]) =
+          wdlParams.path -> wdlParams.inputs.map(fileParameter => fileParameter.name -> fileParameter.typedValue).toMap
 
-        def toProjectFileConfiguration(path: Path, inputs: Map[String, TypedValue]): ProjectFileConfiguration =
-          ProjectFileConfiguration(path, inputs.map((FileParameter.apply _).tupled).toList)
+        def toWdlParams(path: Path, inputs: Map[String, TypedValue]): WdlParams =
+          WdlParams(path, inputs.map((FileParameter.apply _).tupled).toList)
 
         getLastByProjectId(projectConfiguration.projectId, userId).flatMap {
           case Some(config) =>
-            val oldFileConfigsMap = config.projectFileConfigurations.map(toTuple).toMap
-            val newFileConfigsMap = projectConfiguration.projectFileConfigurations.map(toTuple).toMap
+            val oldWdlParamsMap = List(config.wdlParams).map(toTuple).toMap
+            val newWdlParamsMap = List(projectConfiguration.wdlParams).map(toTuple).toMap
```

Comment:
This definitely can be simplified. Why do you need to convert single configuration to List and then to tuple and then to Map if you know that you have only one configuration and its path is known?

Reply:
I believe that that whole logic may be redundant here from now on. And just
```
def addConfiguration(projectConfiguration: ProjectConfiguration, userId: UserId): Future[Unit] =
        repository.addConfiguration(projectConfiguration)
```
will do just fine. Am I right?

Reply:
Tried to refactor this a little